frontend/src/views/Game.tsx

```diff
@@ -147,8 +148,16 @@ function GamePage() {
     setTimeUp(newGame.gameTimer.timeUp);
   };
 
+  const setCurrentLanguage = (newLanguage: Language) => {
+    currentLanguage[currentProblem] = newLanguage;
+  };
+
+  const setCurrentCode = (newCode: string) => {
+    currentCode[currentProblem] = newCode;
+  };
```

Comment:
With react state you're [required](https://stackoverflow.com/questions/37755997/why-cant-i-directly-modify-a-components-state-really) to use the provided `setX` methods that they provide, any state changes have to come through those methods (so you'll have to revert back to `const [currentLanguage, setCurrentLanguages] = useState(...)`. So in this situation, I'd recommend taking a look at `ProblemDisplay.tsx` starting at about line 220 to get an idea for how to modify lists and try to mimic that here

Idk if this is related to the issue you're currently having but hopefully changing this could potentially help?